services/streammanager/kafka/kafkamanager.go

```diff
@@ -414,11 +469,17 @@ func prepareBatchOfMessages(topic string, batch []map[string]interface{}, timest
 			pkgLogger.Errorf("batch from topic %s is missing the userId attribute", topic)
 			continue
 		}
-
 		marshalledMsg, err := json.Marshal(message)
 		if err != nil {
 			return nil, err
 		}
+		codec := p.getCodecs()
+		if codec != nil {
```

Comment:
Instead of checking for not `nil` we can use `len` and cover empty non-nil slices as well:
`if len(codecs) > 0 {`. FYI `len` returns `0` if the slice is `nil`.